document_ocr/models/ir_attachment.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+import subprocess
+from PIL import Image
+from StringIO import StringIO
+from openerp import api, models
+
+_logger = logging.getLogger(__name__)
+_marker_phrase = '[[waiting for OCR]]'
+
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    @api.model
+    def _index(self, data, datas_fname, file_type):
+        mimetype, content = super(IrAttachment, self)._index(
+            data, datas_fname, file_type)
+        if not content or content == 'image':
+            if self.env['ir.config_parameter'].get_param(
+                'document_ocr.synchronous', 'False'
+            ) == 'True' or self.env.context.get('document_ocr_force'):
+                content = self._index_ocr(mimetype, data, datas_fname,
+                                          file_type)
+            else:
+                content = _marker_phrase
+
+        return mimetype, content
+
+    @api.model
+    def _index_ocr(self, mimetype, data, datas_fname, file_type):
+        dpi = int(
+            self.env['ir.config_parameter'].get_param(
+                'document_ocr.dpi', '500'))
+        top_type, sub_type = mimetype.split('/', 1)
+        if hasattr(self, '_index_ocr_get_data_%s' % sub_type):
+            image_data = getattr(self, '_index_ocr_get_data_%s' % sub_type)(
+                data, datas_fname, file_type, dpi)
+        else:
+            image_data = StringIO()
+            try:
+                Image.open(StringIO(data)).save(image_data, 'tiff',
+                                                dpi=(dpi, dpi))
+            except IOError:
+                _logger.exception('Failed to OCR image')
+                return None
+        process = subprocess.Popen(
```

Comment:
Mainly just line reduction and more Pythonic usage. It's just a shell wrapper though, so beyond developer usability technically nothing.
```
image_to_string(Image.open(), lang='eng')
```